src/modules/commander/Commander.cpp

```diff
@@ -3950,19 +3952,26 @@ void Commander::data_link_check(bool &status_changed)
 
 	// AVOIDANCE SYSTEM state check (only if it is enabled)
 	if (_obs_avoid.get()) {
+		static bool print_once = false;
 
 		//if avoidance never started
 		if (_datalink_last_heartbeat_avoidance_system == 0 && hrt_elapsed_time(&_avoidance_system_not_started) > 5_s) {
 			_avoidance_system_not_started = hrt_absolute_time();
-			mavlink_log_critical(&mavlink_log_pub, "AVOIDANCE SYSTEM DID NOT START");
+			mavlink_log_critical(&mavlink_log_pub, "WAITING FOR AVOIDANCE SYSTEM");
 		}
 
 		//if heartbeats stop
 		if ((_datalink_last_heartbeat_avoidance_system > 0)
 		    && (hrt_elapsed_time(&_datalink_last_heartbeat_avoidance_system) > 5_s) &&
 		    (hrt_elapsed_time(&_avoidance_system_lost) > 5_s)) {
 			_avoidance_system_lost = hrt_absolute_time();
-			mavlink_log_critical(&mavlink_log_pub, "AVOIDANCE SYSTEM LOST");
+
+			if (!print_once) {
```

Comment:
I think there is no need of `print_once` if you set `_avoidance_system_lost=true`